src/Bonsai.ML.Torch/Diagonal.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.ComponentModel;
+using System.Reactive.Linq;
+using System.Xml.Serialization;
+using static TorchSharp.torch;
+
+namespace Bonsai.ML.Torch
+{
+    /// <summary>
+    /// Creates a diagonal matrix. If input is a 1D tensor, it creates a diagonal matrix with the elements of the tensor on the diagonal.
+    /// If input is a 2D tensor, it returns the diagonal elements as a 1D tensor.
+    /// </summary>
+    [Combinator]
+    [ResetCombinator]
+    [Description("Creates a diagonal matrix with the given data type, size, and value.")]
+    [WorkflowElementCategory(ElementCategory.Source)]
+    public class Diagonal
+    {
+        /// <summary>
+        /// The input matrix.
+        /// </summary>
+        [Description("The input matrix.")]
+        [TypeConverter(typeof(UnidimensionalArrayConverter))]
+        public double[] Input { get; set; }
+
+        /// <summary>
+        /// The data type of the tensor elements.
+        /// </summary>
+        [Description("The data type of the tensor elements.")]
+        public ScalarType? Type { get; set; }
+
+        /// <summary>
+        /// The device on which to create the tensor.
+        /// </summary>
+        [Description("The device on which to create the tensor.")]
+        [XmlIgnore]
+        public Device Device { get; set; } = null;
+
+        /// <summary>
+        /// The diagonal offset. Default is 0, which means the main diagonal.
+        /// </summary>
+        [Description("The diagonal offset. Default is 0, which means the main diagonal.")]
+        public int Offset { get; set; } = 0;
+
+        /// <summary>
+        /// Creates a diagonal matrix.
```

Comment:
```suggestion
        /// Creates an observable sequence containing a single diagonal matrix constructed from the specified data type, size and values.
```

Editing for consistency since from the point of view of C# all operator methods always return an observable sequence (even if it contains only a single element).

Also noting sometimes you add documentation to the input / output parameters, and sometimes they are omitted, but we can revisit this later and do a general review of docs.